lib/vehicle.rb

```diff
@@ -1,25 +1,41 @@
 require 'date'
 
 class Vehicle
-  attr_reader :vin,
-              :year,
-              :make,
-              :model,
-              :engine
+  attr_reader :vin, :year, :make, :model, :engine, :plate_type
+  attr_accessor :registration_date
+              
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @plate_type = calculate_plate_type
+    @registration_date =vehicle_details[:registration_date]
   end
 
   def antique?
     Date.today.year - @year > 25
   end
 
-  def electric_vehicle?
+  def electric?
     @engine == :ev
   end
+
+  def plate_type
+    calculate_plate_type
```

Comment:
So you've created a method here called `plate_type`, however your attr_reader has also exposed a method called `plate_type`, so you essentially have 2 methods called the same thing. I would just get rid of lines 26-28, set @plate_type on line 14 to nil, and then call `calculate_plate_type` from the Facility class when a vehicle is being registered. Or you can have it called here right away upon initialization.